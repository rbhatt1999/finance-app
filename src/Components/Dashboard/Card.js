import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col, Container } from 'react-bootstrap';
import { BsArrowRightCircle, BsArrowUp, BsArrowDown } from 'react-icons/bs';
import { NavLink } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import graph from '../../images/chart-graph.png';
import styles from '../Pages/dashboard.module.css';
import { fetchStockDetails } from '../../Redux/stocks/stocksThunks';

const Card = ({
  change, companyName, price, changesPercentage, symbol,
}) => {
  const dispatch = useDispatch();

  const onClick = () => {
    dispatch(fetchStockDetails({ symbol, companyName, price }));
  };

  return (
    <Col className={styles.card} onClick={onClick}>
      <div>
        <NavLink to="/details">
          <BsArrowRightCircle className={styles.direct} onClick={onClick} />
        </NavLink>
        <span>
          <b>{companyName}</b>
        </span>
      </div>
      <div className="top-card">
        <span>
          <h6>{symbol}</h6>
        </span>
        <span>
          <b>
            $
            {price}
          </b>
        </span>
        <Container fluid>
          <Row>
            <Col>
              <div>
                <span>
                  <BsArrowUp color="green" />
                  {changesPercentage}
                </span>
                <span>
                  <BsArrowDown color="red" />
                  {change}
                </span>
              </div>
            </Col>
            <Col>
              <img src={graph} className={styles.chart} alt="chart" />
            </Col>
          </Row>
        </Container>
      </div>
    </Col>
  );
};

export default Card;

Card.propTypes = {
  change: PropTypes.number.isRequired,
  companyName: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  changesPercentage: PropTypes.number.isRequired,
  symbol: PropTypes.string.isRequired,
};
