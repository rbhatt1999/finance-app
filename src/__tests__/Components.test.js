import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import Card from '../Components/Dashboard/Card';
import store from '../Redux/configureStore';

describe('the card items', () => {
  it('should show the card data', () => {
    const { asFragment } = render(<Provider store={store}><Card change={4.4} companyName="Google" price={5000} changesPercentage={3.76} symbol="G" /></Provider>, { wrapper: MemoryRouter });
    expect(asFragment()).toMatchSnapshot();
  });
});
