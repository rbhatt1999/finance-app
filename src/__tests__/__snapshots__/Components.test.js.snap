// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the card items should show the card data 1`] = `
<DocumentFragment>
  <div
    class="card col"
  >
    <div>
      <a
        class=""
        href="/details"
      >
        <svg
          class="direct"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 8a7 7 0 1 0 14 0A7 7 0 0 0 1 8zm15 0A8 8 0 1 1 0 8a8 8 0 0 1 16 0zM4.5 7.5a.5.5 0 0 0 0 1h5.793l-2.147 2.146a.5.5 0 0 0 .708.708l3-3a.5.5 0 0 0 0-.708l-3-3a.5.5 0 1 0-.708.708L10.293 7.5H4.5z"
            fill-rule="evenodd"
          />
        </svg>
      </a>
      <span>
        <b>
          Google
        </b>
      </span>
    </div>
    <div
      class="top-card"
    >
      <span>
        <h6>
          G
        </h6>
      </span>
      <span>
        <b>
          $5000
        </b>
      </span>
      <div
        class="container-fluid"
      >
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div>
              <span>
                <svg
                  color="green"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: green;"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 15a.5.5 0 0 0 .5-.5V2.707l3.146 3.147a.5.5 0 0 0 .708-.708l-4-4a.5.5 0 0 0-.708 0l-4 4a.5.5 0 1 0 .708.708L7.5 2.707V14.5a.5.5 0 0 0 .5.5z"
                    fill-rule="evenodd"
                  />
                </svg>
                3.76
              </span>
              <span>
                <svg
                  color="red"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  style="color: red;"
                  viewBox="0 0 16 16"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M8 1a.5.5 0 0 1 .5.5v11.793l3.146-3.147a.5.5 0 0 1 .708.708l-4 4a.5.5 0 0 1-.708 0l-4-4a.5.5 0 0 1 .708-.708L7.5 13.293V1.5A.5.5 0 0 1 8 1z"
                    fill-rule="evenodd"
                  />
                </svg>
                4.4
              </span>
            </div>
          </div>
          <div
            class="col"
          >
            <img
              alt="chart"
              class="chart"
              src="chart-graph.png"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
